app/Platform/Data/GameTopAchieverData.php

```diff
@@ -5,17 +5,34 @@
 namespace App\Platform\Data;
 
 use App\Data\UserData;
+use Carbon\Carbon;
 use Spatie\LaravelData\Data;
 use Spatie\TypeScriptTransformer\Attributes\TypeScript;
 
 #[TypeScript('GameTopAchiever')]
 class GameTopAchieverData extends Data
 {
     public function __construct(
-        public int $rank,
-        public UserData $user,
-        public int $score,
-        public ?PlayerBadgeData $badge,
+        public int $userId,
+        public ?UserData $user,
+        public int $achievementsUnlockedHardcore,
+        public int $pointsHardcore,
+        public Carbon $lastUnlockHardcoreAt,
+        public ?Carbon $beatenHardcoreAt,
     ) {
     }
+
+    public static function fromTopAchiever(array $topAchiever): self
+    {
+        return new self(
+            userId: $topAchiever['user_id'],
+            user: UserData::from($topAchiever['user']),
```

Comment:
> Undefined array key "user"

http://localhost:64000/event/21

Strangely, I get a similar error accessing the backing game page:

http://localhost:64000/game/6189

Reply:
I seem to be unable to repro this locally or on stage https://stage.retroachievements.org/game/6189.

Possible something could be wrong in your local DB?

Reply:
It's a caching issue. Check out `release`. Visit a random game page. Check out this branch. Refresh the page. The cached data doesn't have the `user` subrecord added in #3279. https://github.com/RetroAchievements/RAWeb/pull/3279/files#diff-91f62e6ba07e1193c252fed5dc40e57518c9d09705b12119af186956ffadfaa0R138

Reply:
Ah, now it makes sense.

I think the right solution here is probably to change the cache key. Updated in latest.

Reply:
This does fix the error, but I can't help but wonder if caching the whole user record will lead to some sort of stale data showing up in the list. Most likely the `display_name`. I think the card is dynamically generated on mouseover.

Which fields from user are actually needed?

Reply:
I think we only really need the user display name and avatar URL.

Updated in latest to only send those two values to the UI.